File: tests/Generator/resource/rust/oop/human_type.rs
use serde::{Serialize, Deserialize};
use human_type::HumanType;

#[derive(Serialize, Deserialize)]
pub struct HumanType {
    #[serde(rename = "firstName")]
    first_name: Option<String>,

    #[serde(rename = "parent")]
    parent: Option<HumanType>,

}

File: tests/Generator/resource/python/oop/human_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .human_type import HumanType


class HumanType(BaseModel):
    first_name: Optional[str] = Field(default=None, alias="firstName")
    parent: Optional[HumanType] = Field(default=None, alias="parent")
    pass


